app/src/main/java/ch/epfl/skysync/ui/components/FlightForm.kt

```diff
@@ -0,0 +1,666 @@
+                                          addVehicle = false
+                                        },
+                                        text = { Text(item.name) })
+                                  }
+                                }
+                              }
+                        }
+                  }
+                }
+                // Drop down menu for the balloon
+                item {
+                  val balloonTitle = "Balloon"
+                  Text(
+                      modifier = Modifier.padding(horizontal = defaultPadding),
+                      text = balloonTitle,
+                      style = MaterialTheme.typography.headlineSmall)
+                  MenuDropDown(
+                      defaultPadding = defaultPadding,
+                      title = balloonTitle,
+                      value = balloonValue,
+                      onclickMenu = { item -> balloonValue = item },
+                      items = balloons)
+                }
+                // Drop down menu for the basket
+                item {
+                  val basketTitle = "Basket"
+                  Text(
+                      modifier = Modifier.padding(horizontal = defaultPadding),
+                      text = basketTitle,
+                      style = MaterialTheme.typography.headlineSmall)
+                  MenuDropDown(
+                      defaultPadding = defaultPadding,
+                      title = basketTitle,
+                      value = basketValue,
+                      onclickMenu = { item -> basketValue = item },
+                      items = baskets)
+                }
+              }
+          // Button to add the flight to the list of flights
+          Button(
+              modifier = Modifier.fillMaxWidth().padding(defaultPadding).testTag("$title Button"),
+              onClick = {
+                nbPassengersValueError =
+                    nbPassengersValue.isEmpty() || nbPassengersValue.toInt() <= 0
+                flightTypeValueError = flightTypeValue == null
+                isError = nbPassengersValueError || flightTypeValueError
+                if (!isError) {
+                  val vehicles: List<Vehicle> =
+                      if (vehicle == null) emptyList() else listOf(vehicle!!)
+                  val team = Team(crewMembers.toList())
+                  val newFlight =
+                      PlannedFlight(
+                          nPassengers = nbPassengersValue.toInt(),
+                          date = dateValue,
+                          flightType = flightTypeValue!!,
+                          timeSlot = timeSlotValue,
+                          balloon = balloonValue,
+                          basket = basketValue,
+                          vehicles = vehicles,
+                          team = team,
+                          id = currentFlight?.id ?: "unset id")
+                  flights.add(newFlight)
```

Comment:
Instead of taking a mutable list of flights as argument, consider taking a `onAddFlight`/`onModifyFlight` callback as argument (unless there is a specific reason to take a list of flights ?).